src/sentry/rules/conditions/level.py

```diff
@@ -58,7 +59,13 @@ def passes(self, event, state, **kwargs):
             return False
 
         desired_level = int(desired_level)
-        level = int(event.group.level)
+        # Fetch the event level from the tags first since event.level is
+        # event.group.level which may not be correct
+        level_tag = event.get_tag('level')
+        if level_tag and level_tag in LOG_LEVEL_REVERSE_MAP:
```

Comment:
This could all be simplified now to just:

```python
try:
    level = LOG_LEVEL_REVERSE_MAP[event.get_tag('level')]
except KeyError:
    return False
```

which will be slightly faster for the normal case. I don't think we'd want to actually raise an exception here, we'd just want to say the rule doesn't match if there's no log level somehow.